File: JZiOS/JZiOS/Session.swift
import Foundation
import RealmSwift
import ObjectMapper

class SessionResult: Mappable {
    var sessions: [Session]?
    
    required init?(map: Map) {
    }
    
    func mapping(map: Map) {
        sessions    <- map["sessions"]
    }
}

class Session: Object, Mappable {
    var intendedAudience: String?
    var endTimeZulu: String?
    var keywords: Array<String>?
    var format: String?
    var language: String?
    var sessionId: String?
    var abstract: String?
    var published: String?
    var video: String?
    var title: String?
    var room: String?
    var conferenceId: String?
    var startTimeZulu: String?
    var speakers: Array<Speaker>?
    var startTime: String?
    var endTime: String?
    var slug: String?
    
    required convenience init?(map: Map) {
        self.init()
    }
    
    func mapping(map: Map) {
        intendedAudience    <- map["intendedAudience"]
        endTimeZulu         <- map["endTimeZulu"]
        keywords            <- map["keywords"]
        format              <- map["format"]
        language            <- map["language"]
        sessionId           <- map["sessionId"]
        abstract            <- map["abstract"]
        published           <- map["published"]
        video               <- map["video"]
        title               <- map["title"]
        room                <- map["room"]
        conferenceId        <- map["conferenceId"]
        startTimeZulu       <- map["startTimeZulu"]
        speakers            <- map["speakers"]
        startTime           <- map["startTime"]
        endTime             <- map["endTime"]
        slug                <- map["slug"]
        
    }
}

class Speaker: Mappable {
    var pictureUrl: String?
    var name: String?
    var bio: String?
    
    required init?(map: Map) {
    }
    
    func mapping(map: Map) {
        pictureUrl      <- map["pictureUrl"]
        name            <- map["name"]
        bio             <- map["bio"]
    }
}

File: JZAndroid/app/src/main/java/no/schedule/javazone/v3/io/model/Session.java
package no.schedule.javazone.v3.io.model;

import com.google.gson.annotations.SerializedName;

import java.util.Random;

public class Session {
  @SerializedName("sessionId")
  public String id;
  @SerializedName("intendedAudience")
  public String intendedAudience;
  @SerializedName("keywords")
  public String[] tags;
  public String endTimeZulu;
  public String level;
  public String length;
  public String language;
  @SerializedName("abstract")
  public String description;
  public String published;
  public String video;
  public String title;
  public String room;
  public String conferenceId;
  public String startTimeZulu;
  public Speaker[] speakers;
  public String startTime;
  public String endTime;
  public String slug;
  public transient boolean isLivestream;


  public class RelatedContent {
    public String id;
    public String name;

    @Override
    public String toString() {
      return "RelatedContent{" +
          "id='" + id + '\'' +
          ", name='" + name + '\'' +
          '}';
    }
  }

  public String makeTagsList() {
    int i;
    if (tags == null || tags.length == 0) return "";
    StringBuilder sb = new StringBuilder();
    sb.append(tags[0]);
    for (i = 1; i < tags.length; i++) {
      sb.append(",").append(tags[i]);
    }
    return sb.toString();
  }

  public boolean hasTag(String tag) {
    for (String myTag : tags) {
      if (myTag.equals(tag)) {
        return true;
      }
    }
    return false;
  }

  public String getImportHashCode() {
    return (new Random()).nextLong()+"";
  }
}
